frontend/packages/ceph-storage-plugin/src/components/dashboards/odf-system-dashboard.tsx

```diff
@@ -24,11 +26,16 @@ const ODFSystemDashboard: React.FC<ODFSystemDashboardPageProps> = ({
   ...rest
 }) => {
   const { t } = useTranslation();
+  const isObjectServiceAvailable = useFlag(MCG_FLAG);
+  const isCephAvailable = useFlag(CEPH_FLAG);
   const { systemName } = rest.match.params;
+  const dashboardTab = isCephAvailable === false && isObjectServiceAvailable ? OBJECT : BLOCK_FILE;
+  const defaultDashboard = React.useRef(dashboardTab);
```

Comment:
Do we really need to `useRef`? 
Isn't dashboardTab enough?

Reply:
It will cause issue if I will navigate to the dashboard page before `isObjectServiceAvailable` flag value is set.

`isObjectServiceAvailable` takes some time to get `true` when compared to other flags. If I navigate to dashboard before that then dashboardTab value will be set to `block-file` and it will update the pathname as well to `/block-file`. Now, even if `isObjectServiceAvailable` is set to `true` later it will not navigate to object tab automatically as `useEffect` currently only modifies the pathname if it endswith `systemName or 'overview'` : https://github.com/openshift/console/blob/1d0fc9c40c73260d837a7c08dace9f559a490554/frontend/packages/ceph-storage-plugin/src/components/dashboards/odf-system-dashboard.tsx#L65

Reply:
It is an edge case when someone reloads the page and navigate to system overview dashboard before `MCG flag` value is set to `true`.